lib/spack/spack/test/cmd/url.py

```diff
@@ -77,51 +72,64 @@ def test_version_parsed_correctly():
     assert not version_parsed_correctly(MyPackage('', ['0.18.0']), 'oce-0.18.0')   # noqa
 
 
-def test_url_parse(parser):
-    args = parser.parse_args(['parse', 'http://zlib.net/fossils/zlib-1.2.10.tar.gz'])
-    url(parser, args)
+def test_url_parse():
+    url('parse', 'http://zlib.net/fossils/zlib-1.2.10.tar.gz')
 
 
-@pytest.mark.xfail
-def test_url_parse_xfail(parser):
+def test_url_with_no_version_failes():
     # No version in URL
-    args = parser.parse_args(['parse', 'http://www.netlib.org/voronoi/triangle.zip'])
-    url(parser, args)
+    with pytest.raises(UndetectableVersionError):
+        url('parse', 'http://www.netlib.org/voronoi/triangle.zip')
 
 
-def test_url_list(parser):
-    args = parser.parse_args(['list'])
-    total_urls = url_list(args)
```

Comment:
It's definitely still possible.  I still call some directly below.  Do you want these changed back?  Or should we do both?  It's hard to say which level of testing (command or sub-function) is the right one.  It's d be nice to test both as some of the corner cases in Python 3 porting and other spots arose in commands.